### ISSUE TITLE
Experimental branch body parsing, Postgres and TRPC errors don't work

### ISSUE THREAD

**Issue by pellicceama** [opennextjs/opennextjs-cloudflare#205]
### Describe the bug

We tried the experimental branch to use middleware and stumbled into significant issues. 

TLDR; 
1. Database queries using PostgresJS get stuck with no response (only works with node-pg + patches), 
2. Body parsing in API routes returns 500 errors, 
3. tRPC error responses return HTML 500 pages instead of proper JSON error objects. 

Tested with @opennextjs/cloudflare@2ab976f and Next.js 14.2.13.

### Steps to reproduce

We created a minimal reproduction example with more details here: https://github.com/openintegrations/opennext-cf-sample 

### Expected behavior

The mentioned issues should work. 

### @opennextjs/cloudflare version

experimental @opennextjs/cloudflare@2ab976f

### Wrangler version

3.95.0

### next info output

```shell
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.6.0: Fri Jul  5 17:55:37 PDT 2024; root:xnu-10063.141.1~2/RELEASE_ARM64_T6030
  Available memory (MB): 18432
  Available CPU cores: 11
Binaries:
  Node: 22.9.0
  npm: 10.8.3
  Yarn: N/A
  pnpm: 9.13.2
Relevant Packages:
  next: 14.2.13 // An outdated version detected (latest is 15.1.2), upgrade is highly recommended!
  eslint-config-next: N/A
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.7.2
Next.js Config:
  output: N/A
 ⚠ An outdated version detected (latest is 15.1.2), upgrade is highly recommended!
   Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
   Read more - https://nextjs.org/docs/messages/opening-an-issue
```

**Comment by pellicceama** [comment]
@vicb FYI we managed to make it work regardless on our fairly complex next app https://github.com/openintegrations/openint/tree/cf-new

We are part of the Cloudflare Worker accelerator, our previous name was Evefan. Feel free to ping me on the discord or over email if we can be helpful :)

**Comment by vicb** [comment]
> We tried the experimental branch to use middleware and stumbled into significant issues.

The experimental branch has been merged back to main and release as 0.3.x.
Docs are up to date at opennext.js.org/cloudflare

> FYI we managed to make it work regardless on our fairly complex next app https://github.com/openintegrations/openint/tree/cf-new

What works/does not work in the end?
Do you still need help with anything?
Sorry I'm a little confused about the status here

**Comment by pellicceama** [comment]
>The experimental branch has been merged back to main and release as 0.3.x.
Ahh great to hear! Just upgraded [our minimal example](https://github.com/openintegrations/opennext-cf-sample) to it and that seems to fix the body parsing issue. 

> What works/does not work in the end?

We're OK, no immediate ask for help just thought I'd report these issues as we had to do a [number of monkey patches](https://github.com/openintegrations/openint/blob/cf-new/apps/web/patch-cloudflare.bin.js) to make our app work. 

The main thing to call your attention to right now is the issue with PostgresJS, which I know is a key dependency for CF. I've also reported [an issue to node-pg](https://github.com/brianc/node-postgres/issues/3349#issuecomment-2561969479).

I also upgraded our cf-new main app branch to 0.3.2 and now see that Clerk middleware that used to work now isn't working. Need to investigate further though.

**Comment by vicb** [comment]
Are there still issues with the latest release together with the latest wrangler?

**Comment by pellicceama** [comment]
We haven't upgraded yet, feel free to close this and i can reopen if they're there when we get to it as we don't currently have a timeline for it given that our patches work.

**Comment by vicb** [comment]
Let's do that. Thanks!
